### PR TITLE
Use FUSE with FileField URLs

### PR DESCRIPTION
**Pull request by banesullivan**
This adds a fix to be able to bypass the `field_file_to_local_path` for S3FileField which was a limitation of #257. Currently, we fallback to vsicurl for FileFields where we can 

The trick was making sure the host on the file URLs were correct in dev/prod environments (e.g. `minio` hostname in a dev environment)

This also adds a general test to make sure we can use FUSE with FileField (when run through docker-compose where FUSE is installed... CI does not test this).

### CODE REVIEW

**Comment by banesullivan** [comment]
This is still failing in local development. It's as though the credential in the URL confuse rasterio??

```
Traceback (most recent call last):
  File "/opt/django-project/rgd/geodata/tasks.py", line 15, in _safe_execution
    func(*args, **kwargs)
  File "/opt/django-project/rgd/geodata/models/imagery/etl.py", line 92, in read_image_file
    _read_image_to_entry(image_entry, file_path)
  File "/opt/django-project/rgd/geodata/models/imagery/etl.py", line 55, in _read_image_to_entry
    with rasterio.open(image_file_path) as src:
  File "/opt/django-project/.tox/test/lib/python3.8/site-packages/rasterio/env.py", line 435, in wrapper
    return f(*args, **kwds)
  File "/opt/django-project/.tox/test/lib/python3.8/site-packages/rasterio/__init__.py", line 207, in open
    s = DatasetReader(path, driver=driver, sharing=sharing, **kwargs)
  File "rasterio/_base.pyx", line 262, in rasterio._base.DatasetBase.__init__
rasterio.errors.RasterioIOError: '/tmp/rgd/http/minio:9000/test-django-storage/81368dbb-7ccf-422c-a007-339aabceebec/Elevation.tif?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minioAccessKey%2F20210308%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210308T063740Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=1be97f3cfa5fa20423ac86209c460fd5c986882e7d283b0b30c778bcc9805da4..' not recognized as a supported file format.
```

**Comment by banesullivan** [comment]
For now, using `vsicurl` to load files in the minio/s3 app storage works just as fine (until we need anything other than GDAL to load the files). For that reason, I may revert some of these changes and merge in others.

File: rgd/geodata/models/common.py [hunk 1]
```diff
@@ -225,7 +225,9 @@ def yield_local_path(self, vsi=False):
     def get_url(self):
         """Get the URL of the stored resource."""
         if self.type == FileSourceType.FILE_FIELD:
-            return self.file.url
+            with patch_internal_presign(self.file):
+                url = self.file.url
+            return url
```
Comment (banesullivan) [inline rgd/geodata/models/common.py:230]: #292 will take precedent

**Comment by banesullivan** [comment]
It seems this error reading files with credential in the URL is actually coming from GDAL:

```py
# Path is a FUSE 
ds = gdal.Open(path)
ERROR 4: foo/http/localhost:9000/django-storage/0a48e957-ea4f-40bf-b15a-88d2661b3497/LC08_L1TP_029040_20210107_20210107_01_RT_B4.tiff?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Credential=minioAccessKey%2F20210308%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Date=20210308T164037Z&X-Amz-Expires=604800&X-Amz-SignedHeaders=host&X-Amz-Signature=54f060b9776a2c206d80a9ca5bebde0225305a57cb8c24d4f4cd0b818f8311d4..: File name too long

# Or with vsi
In [16]: gdal.Open(vsicurl)
ERROR 4: `/vsicurl?url=http%3A%2F%2Flocalhost%3A9000%2Fdjango-storage%2F0a48e957-ea4f-40bf-b15a-88d2661b3497%2FLC08_L1TP_029040_20210107_20210107_01_RT_B4.tiff%3FX-Amz-Algorithm%3DAWS4-HMAC-SHA256%26X-Amz-Credential%3DminioAccessKey%252F20210308%252Fus-east-1%252Fs3%252Faws4_request%26X-Amz-Date%3D20210308T164037Z%26X-Amz-Expires%3D604800%26X-Amz-SignedHeaders%3Dhost%26X-Amz-Signature%3D54f060b9776a2c206d80a9ca5bebde0225305a57cb8c24d4f4cd0b818f8311d4&use_head=no&list_dir=no' not recognized as a supported file format.
```

I either get `File name too long` or `not recognized as a supported file format` errors

**Comment by jbeezley** [comment]
You might try some of the config options mentioned here https://github.com/OSGeo/gdal/issues/1609#issuecomment-498199300.  At the very least, some of the debug options could be enlightening.

**Pull request closed without merging**